Added new country test (#1)
tysonwolker/front-end-challenge: README.md

The requirements are intentionally sparse (the devil is in the detail). We don't want you to do a lot, but what you do should be your best work and should clearly demonstrate you embody the [Three Virtues](http://threevirtues.com/).

## Table of contents

- [The rules](#the-rules)
- [The challenge](#the-challenge)
- [The requirements](#the-requirements)
  - [Countries page](#countries-page)
  - [Country page](#country-page)
- [Additional information](#additional-information)
  - [Assets](#assets)
- [Contributing](#contributing)
- [Credits](#credits)
- [License](#license)

## The rules

> - Read and understand [The Challenge](#the-challenge)

Reviewer: ```suggestion
- Read and understand [The challenge](#the-challenge)
```

> - Create solutions that satisfy [The Requirements](#the-requirements)

Reviewer: ```suggestion
- Create solutions that satisfy [The requirements](#the-requirements)
```

- Send us a link to the repository containing your solution once you are done

## The challenge

We want to build an app that can display a list of countries with the capability of filtering results through a search mechanism, paginating countries and showing information about each individual country.

> Using the following endpoint to [get the list of all countries](https://restcountries.eu/rest/v2/all).

Reviewer: nit: "Using the following" reads a little strangely when the link is part of the text like this. How about:
```suggestion
Use the [RESTCountries endpoint](https://restcountries.eu/rest/v2/all) to get the list of all countries.
```

## The Requirements

### Countries Page

The first feature is to create a page where a customer can view all the countries that are returned from the countries API. Create a web application using [React](https://reactjs.org/) that satisfies the following scenarios:

```text
Scenario 1: Countries loading
When I visit the Countries page
Then I should see a loading spinner

Scenario 2: Countries loaded
Given I am on the countries page
When the countries have finished loading
Then I should see the first 10 countries in alphabetical order
And display their country name

Scenario 3: Click previous page
Given I am on the Countries page
And I am not already on the first page
When I click on the previous button
Then I should see the previous 10 countries in alphabetical order

Scenario 4: Click next page
Given I am on the Countries page
And I am not already on the last page
When I click on the next button
Then I should see the next 10 countries in alphabetical order

Scenario 5: Search
Given I am on the Countries page
When I enter text in the search text input
And when I click search

> Then I should check if the country's name contains the search text

Reviewer: It's not clear to me what this means. Can we rephrase it in terms of what the user will see?

Reviewer 2: Agree. There are a few here that need rewriting from the customers perspective.

"And display their country name"
"Then hide the next page button"
"Then hide the previous page button"
"Then take me to the country page"
etc...

## The Rules

- Read and understand [The Challenge](#the-challenge)
- Create solutions that satisfy [The Requirements](#the-requirements)
- Send a link to the repository containing your solution to careers@q-ctrl.com
- Write your own pagination logic
- Write your own search logic, you are not marked on how advanced your search algorithm is
- Try not to spend more than 3 hours
- You are free to use a UI library such as [Material-UI](https://material-ui.com/)

## The Challenge

We want to build an app that can display a list of countries with the capability of filtering results through a search mechanism, paginating countries and showing information about each individual country.

Using the following endpoint to [get the list of all countries](https://restcountries.eu/rest/v2/all).

## The Requirements

### Countries Page

The first feature is to create a page where a customer can view all the countries that are returned from the countries API. Create a web application using [React](https://reactjs.org/) that satisfies the following scenarios:

```text
Scenario 1: Countries loading
When I visit the Countries page
Then I should see a loading spinner

Scenario 2: Countries loaded

> Given I am on the countries page

Reviewer 2: "countries page" or "Countries page". Choose one. Use consistently.

  - [The Rules](#the-rules)
  - [The Challenge](#the-challenge)
  - [The Requirements](#the-requirements)
    - [Countries Page](#countries-page)
    - [Country Page](#country-page)
  - [Additional Information](#additional-information)
    - [Assets](#assets)
  - [Contributing](#contributing)
  - [Credits](#credits)
  - [License](#license)

## The Rules

- Read and understand [The Challenge](#the-challenge)
- Create solutions that satisfy [The Requirements](#the-requirements)
- Send a link to the repository containing your solution to careers@q-ctrl.com
- Write your own pagination logic
- Write your own search logic, you are not marked on how advanced your search algorithm is
- Try not to spend more than 3 hours
- You are free to use a UI library such as [Material-UI](https://material-ui.com/)

## The Challenge

We want to build an app that can display a list of countries with the capability of filtering results through a search mechanism, paginating countries and showing information about each individual country.

Using the following endpoint to get the list of all countries.
https://restcountries.eu/rest/v2/all

## The Requirements

### Countries Page

> The first feature is to create a page where a customer can view all the countries that are returned from the countries API. Create a web application using [React](https://reactjs.org/) that satisfies the following scenarios:

Reviewer: nit: IMO the structure here is a little unclear, because the reference to creating a web application using React is in this "countries page" section, even though it also applies to the "country page". How about changing it to something like:
```
## The Requirements
Create a web application using [React](https://reactjs.org/) consisting of two pages, as describe below.

### Countries Page
The first page allows a customer to view all the countries that are returned from the countries API, and satisfies the following scenarios:
...

### Country Page
The second page allows a customer to view details about an individual country, and satisfies the following scenarios:
...
```
